src/probe_matcher.cpp

```diff
@@ -486,10 +486,14 @@ void ProbeMatcher::list_probes(ast::Program* prog)
           param_lists = get_iters_params(matches);
         else if (probe_type == ProbeType::uprobe)
           param_lists = get_uprobe_params(matches);
+        else if (probe_type == ProbeType::kprobe ||
+                 probe_type == ProbeType::kretprobe)
```

Comment:
```suggestion
        else if (probe_type == ProbeType::kprobe)
```
This only works for kprobes

Reply:
How about we keep it same as fentry/fexit, like:

```
$ sudo bpftrace -lv fentry:vmlinux:do_sys_openat2
fentry:vmlinux:do_sys_openat2
    int dfd
    const char * filename
    struct open_how * how
    int retval
$ sudo bpftrace -lv fexit:vmlinux:do_sys_openat2
fexit:vmlinux:do_sys_openat2
    int dfd
    const char * filename
    struct open_how * how
    int retval
```

After all, kretprobe also needs to know the return value.

```
$ sudo bpftrace -lv kprobe:do_sys_openat2
kprobe:do_sys_openat2
    int do_sys_openat2(int arg0, const char * arg1, struct open_how * arg2);
$ sudo bpftrace -lv kretprobe:do_sys_openat2
kretprobe:do_sys_openat2
    int do_sys_openat2(int arg0, const char * arg1, struct open_how * arg2);
```

right?

Reply:
> After all, kretprobe also needs to know the return value.

For kretprobe and uretprobe, its type is always uint64. The reason it exists for fexit is because you can access the `args` in an fexit probe. I think for `kretprobe` it just might be misleading to list the function signature.